beater/handlers.go

```diff
@@ -106,15 +116,62 @@ func frontendSwitchHandler(feSwitch bool, h http.Handler) http.Handler {
 		if feSwitch {
 			h.ServeHTTP(w, r)
 		} else {
-			sendStatus(w, r, 403, errForbidden)
+			sendStatus(w, r, http.StatusForbidden, errForbidden)
+		}
+	})
+}
+
+func ipRateLimitHandler(rateLimit int, h http.Handler) http.Handler {
+
+	cache, _ := lru.New(rateLimitCacheSize)
+
+	var deny = func(ip string) bool {
+		cache.ContainsOrAdd(ip, rate.NewLimiter(rate.Limit(rateLimit), rateLimit*rateLimitBurstMultiplier))
```

Comment:
I see now that having a separate `Contains` and then `Add` allows us to initialize way fewer rate limiters. This now effectively means we'll create a rate limiter for every request that goes into this handler.  I looked into what the cost of creating those are. I was mostly worried about the `sync.Mutex` that gets created every time, but it looks like it shouldn't be that expensive.

I'm OK with changing back to `Contains` and then `Add`, but also OK to stay with this.

Reply:
that limiter probably goes to the heap. I'll change it back.